tests/shared/src/test/scala/cats/tests/NestedSuite.scala

```diff
@@ -127,11 +127,25 @@ class NestedSuite extends CatsSuite {
 
   {
     // Applicative + ContravariantMonoidal functor composition
-    checkAll("Nested[Option, Const[String, *], *]",
-             ContravariantMonoidalTests[Nested[Option, Const[String, *], *]].contravariantMonoidal[Int, Int, Int]
+//    implicit val isos = Isomorphisms.invariant[Nested[Option, Const[String, *], *]]
+    checkAll(
+      "Nested[Option, Const[String, *], *]",
+      ContravariantMonoidalTests[λ[α => Nested[Option, Const[String, *], α]]].contravariantMonoidal[Int, Int, Int]
+    )
+    checkAll(
+      "ContravariantMonoidal[Nested[Option, Const[String, *], *]",
+      SerializableTests.serializable(ContravariantMonoidal[λ[α => Nested[Option, Const[String, *], α]]])
+    )
+  }
+
+  {
+    // Applicative + Decidable functor composition
+    implicit val eqForOptionPredicateNothing: Eq[Nested[Option, Predicate, Nothing]] = Eq.allEqual
```

Comment:
Experimenting with this for now to see if it passes remotely (not all tests run for me locally [maybe it's my M1?]).
Anyway, it seems not quite right to me, the right instance should be `Eq.by(_.value.isDefined)` or something similar but that does not seem to pass.

Reply:
It's weird, but I guess you can't have `Some[Nothing]`, so the only valid instance of `Option[Nothing]` is `None`, which is a singleton?

Reply:
Yeeeahhhhhh it's kind of "extensionally cool" basically since `Predicate[Nothing]` is essentially `Nothing`. Well, I'll leave it for now, anything stronger is just incompatible with the instance, and the instance makes more sense than the law does really.

Reply:
Oh, right, `Eq.by(_.value.isDefined)` does seem right. CI seems to be passing, does that not pass?

Reply:
I've removed the `right absorption` law. I can't see any way to make it consistent with instances that are reasonable. It's also not a law in any other library implementing this I know of, and the set we have seems comparatively strong.
If it doesn't meet our standards that's alright - maybe someone else will be able to find a reasonable extension to this.